compiler/rustc_mir/src/interpret/step.rs

```diff
@@ -264,10 +268,12 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             NullaryOp(mir::NullOp::SizeOf, ty) => {
                 let ty = self.subst_from_current_frame_and_normalize_erasing_regions(ty);
                 let layout = self.layout_of(ty)?;
-                assert!(
-                    !layout.is_unsized(),
-                    "SizeOf nullary MIR operator called for unsized type"
-                );
+                if layout.is_unsized() {
+                    self.tcx.sess.delay_span_bug(
+                        span,
+                        &format!("SizeOf nullary MIR operator called for unsized type {:?}", ty),
```

Comment:
@oli-obk is this debug string useful? I'm don't know if Debug impl of `TyS` is useful or not.. (perhaps it just prints a type id or something like that?)

Reply:
you can print types with the display impl, which is the human readable form which should totally suffice here

Reply:
Done.